### PR TITLE
Make format and dhall subcommands of dhall executable

### LINKED ISSUE #436: Make `dhall-format` and `dhall-hash` subcommands instead of standalone executables
For command-line consistency, it would be nice for `dhall-format` and `dhall-hash` to become `dhall format` and `dhall hash` (i.e. `format` and `hash` subcommands of the `dhall` executable)

### PR DESCRIPTION
**Pull request by gilligan**
I really just moved code around a bit - fixes #436 

I just realized that Tutorial.hs also has to be adjusted since it is using `dhall-format` and `dhall-hash`. Another thing is with `--inplace` now being a global switch. Is that Ok @Gabriel439 or should that rather be moved down to be a switch to the `format` subcommand?

### CODE REVIEW

**Comment by gilligan** [comment]
Oops, just saw the upstream changes and fixing..

File: src/Dhall/Main.hs [hunk 1]
```diff
@@ -65,6 +68,13 @@ parseOptions = Options <$> parseMode <*> parseExplain <*> parsePlain
             <>  Options.Applicative.help "Disable syntax highlighting"
             )
 
+    parseInplace =
```
Comment (Gabriel439) [inline src/Dhall/Main.hs]: Yeah, I think this should be a `format`-specific option (See https://github.com/dhall-lang/dhall-haskell/pull/451 for an example of how to do that)

Note that it's still worth keeping the intermediate `parseInplace` utility as I imagine that other utilities might reuse it in the future (like the `dhall freeze` command described in https://github.com/dhall-lang/dhall-haskell/issues/437)

Comment (gilligan) [reply src/Dhall/Main.hs]: :ok_hand:

**Comment by gilligan** [comment]
@Gabriel439 i also updated Tutorial.hs, let me know if that works for you.

@fingerzam I suppose your PR was actually ready before mine. I somehow missed it. Sorry. I actually took the format subcommand code from you now but I moved hash and format to their own modules..

**Comment by Gabriel439** [comment]
As I mentioned in #436 I have decided to merge this pull request.  Thanks to both of you, though, for taking the time to do this 🙂

**Pull request merged**